### PR TITLE
Migrate to simple reference resolution and selection

### PR DESCRIPTION
**Pull request by negz**
### Description of your changes

This adds a reference selector for all existing references, and uses the new `managed.APISimpleReferenceResolver` to select and resolve references. See https://github.com/crossplane/crossplane-runtime/pull/151 for more detail.

### Checklist

I have:
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Ensured this PR contains a neat, self documenting set of commits.
- [ ] Updated any relevant [documentation], [examples], or [release notes].
- [ ] Updated the dependencies in [`app.yaml`] to include any new role permissions.

[documentation]: https://github.com/crossplane/crossplane/tree/master/docs
[examples]: https://github.com/crossplane/crossplane/tree/master/cluster/examples
[release notes]: https://github.com/crossplane/crossplane/tree/master/PendingReleaseNotes.md
[`app.yaml`]: https://github.com/crossplane/provider-azure/blob/master/config/stack/manifests/app.yaml

### CODE REVIEW

File: pkg/controller/storage/account/account.go [hunk 1]
```diff
@@ -117,21 +110,6 @@ func (r *Reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 		return reconcile.Result{}, err
 	}
 
-	if !resource.IsConditionTrue(b.GetCondition(runtimev1alpha1.TypeReferencesResolved)) {
```
Comment (hasheddan) [inline pkg/controller/storage/account/account.go:120]: Were we previously resolving references here even though the storage types do not have any ref fields?

Comment (negz) [reply pkg/controller/storage/account/account.go:120]: Yeah, exactly. I figured it made more sense to remove this logic than to port it to the new resolver.

**Comment by negz** [comment]
I've tested this by applying `examples/cache`, `examples/compute`, and `examples/database`. All resources successfully resolved their references and created. I also tested that error messages appeared as expected if I attempted to create a managed resource before the resource it depended on existed, or before the resource it depended on had actually set the referenced value.

This PR decreases test coverage by a little over 1%, presumably because each referencer function under `apis/` previously had tests. I'm not planning on adding tests for the new `ResolveReferences` methods, as they're all effectively boilerplate calls to `reference.NewAPIResolver`, which has its own tests.

**Comment by negz** [comment]
> has that behavior been tested as well?

@hasheddan I didn't test 100% of them, but I have validated that the code generally works as expected by using a selector to pick a resource group reference for a MySQLServer.

**Pull request merged**